crates/net/discv4/src/proto.rs

```diff
@@ -191,6 +194,72 @@ pub struct Packet {
     pub hash: B256,
 }
 
+/// Represents the `from` field in the `Ping` packet
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, RlpEncodableWrapper)]
+pub struct PingNodeEndpoint(pub NodeEndpoint);
```

Comment:
does this have to be pub? we only use this internally?

```suggestion
struct PingNodeEndpoint(NodeEndpoint);
```